stan/math/prim/mat/err/is_pos_definite.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef STAN_MATH_PRIM_MAT_ERR_IS_POS_DEFINITE_HPP
+#define STAN_MATH_PRIM_MAT_ERR_IS_POS_DEFINITE_HPP
+
+#include <stan/math/prim/mat/meta/get.hpp>
+#include <stan/math/prim/mat/meta/length.hpp>
+#include <stan/math/prim/mat/meta/is_vector.hpp>
+#include <stan/math/prim/mat/meta/is_vector_like.hpp>
+#include <stan/math/prim/scal/err/is_positive_size.hpp>
+#include <stan/math/prim/scal/err/is_not_nan.hpp>
+#include <stan/math/prim/mat/err/is_symmetric.hpp>
+#include <stan/math/prim/mat/err/constraint_tolerance.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/mat/fun/value_of_rec.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return <code>true</code> if the matrix is square or if the matrix has
+ * non-zero size, or if the matrix is symmetric, or if it is positive
+ * definite, or if no element is <code>NaN</code>.
+ * @tparam T_y Type of scalar of the matrix, requires class method
+ *   <code>.rows()</code>
+ * @param y Matrix to test
+ * @return <code>true</code> if the matrix is square or if the matrix has non-0
+ *   size, or if the matrix is symmetric, or if it is positive definite, or if
+ *   no element is <code>NaN</code>
+ */
+template <typename T_y>
+inline bool is_pos_definite(const Eigen::Matrix<T_y, -1, -1>& y) {
+  if (is_symmetric(y)) {
```

Comment:
unpack nesting;  combine final bits into single return of `cholesky.info() == Eigen::Success && ...`